lib/ABI/FunctionType/Layout.cpp

```diff
@@ -0,0 +1,963 @@
+/// \file Layout.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <span>
+#include <unordered_set>
+
+#include "revng/ABI/Definition.h"
+#include "revng/ABI/FunctionType/Layout.h"
+#include "revng/ABI/FunctionType/Support.h"
+#include "revng/ABI/FunctionType/TypeBucket.h"
+#include "revng/ADT/SmallMap.h"
+#include "revng/Model/Binary.h"
+#include "revng/Model/Helpers.h"
+#include "revng/Support/Debug.h"
+
+static Logger Log("function-type-conversion-to-raw");
+
+namespace abi::FunctionType {
+
+using RegisterVector = llvm::SmallVector<model::Register::Values, 2>;
+struct DistributedArgument {
+  RegisterVector Registers = {};
+  size_t Size = 0, SizeOnStack = 0;
```

Comment:
I guess my overuse of `std::size_t` is also an early optimization 😅.

Because what if somebody wants to compile that code for 32-bit architecture.

It's not like I care much about the boundaries of the type. If we ever run into a argument with size exceeding 2^32, we have bigger problems than that.

Reply:
P. S. `uint64_t` works, will swap.

Reply:
Yeah, no early optimization plz.

Anything that will end in the model, has to have the same type as in the model.

Reply:
Done.